xwayland/selection/selection.c

```diff
@@ -32,16 +45,63 @@ void xwm_selection_transfer_destroy_property_reply(
 	transfer->property_reply = NULL;
 }
 
-void xwm_selection_transfer_init(struct wlr_xwm_selection_transfer *transfer) {
+void xwm_selection_transfer_init(struct wlr_xwm_selection_transfer *transfer,
+		struct wlr_xwm_selection *selection) {
+	struct wlr_xwm *xwm = selection->xwm;
+
+	transfer->selection = selection;
 	transfer->wl_client_fd = -1;
+
+	transfer->window = xcb_generate_id(xwm->xcb_conn);
+	xcb_create_window(
+		xwm->xcb_conn,
+		XCB_COPY_FROM_PARENT,
+		transfer->window,
+		xwm->screen->root,
+		0, 0,
+		10, 10,
+		0,
+		XCB_WINDOW_CLASS_INPUT_OUTPUT,
+		xwm->screen->root_visual,
+		XCB_CW_EVENT_MASK, (uint32_t[]){
+			XCB_EVENT_MASK_SUBSTRUCTURE_NOTIFY | XCB_EVENT_MASK_PROPERTY_CHANGE
+		}
+	);
+
+	xcb_set_selection_owner(xwm->xcb_conn, transfer->window,
+		xwm->atoms[CLIPBOARD_MANAGER], XCB_TIME_CURRENT_TIME);
```

Comment:
Hm, this doesn't seem right. We should only do this for the outgoing transfer window?

Reply:
Right, this particular copy-paste operation predates my knowing of what `CLIPBOARD_MANAGER` is :)